hn2016_falwa/xarrayinterface.py

```diff
@@ -0,0 +1,534 @@
+        out_fields = _map_collect(
+            lambda field: field._compute_lwa_flux_dirinv(
+                qref=field._temp_dirinv_qref,
+                uref=field._temp_dirinv_u,
+                tref=field._temp_dirinv_tref,
+                fawa=field._temp_dirinv_fawa,
+                ubar=field._temp_dirinv_ubar,
+                tbar=field._temp_dirinv_tbar
+            ),
+            self._fields,
+            ["astarbaro", "ubaro", "urefbaro", "ua1baro", "ua2baro", "ep1baro",
+                "ep2baro", "ep3baro", "ep4", "astar1", "astar2"],
+            postprocess=np.asarray
+        )
+        # Take the first field to extract coordinates and metadata
+        _field = self.fields[0]
+        _nlat = _field.nlat // 2 + _field.nlat % 2
+        # Prepare coordinate-related data for the output:
+        # 1D data, function of latitude only
+        out_dims_y = (*self._other_dims, "ylat")
+        out_shape_y = (*self._other_shape, _nlat)
+        # 2D data, function of longitude and latitude
+        out_dims_xy = (*self._other_dims, "xlon", "ylat")
+        out_shape_xy = (*self._other_shape, _field.xlon.size, _nlat)
+        # 3D data, function of longitude, latitude and height (again, xlon and
+        # plev are currently not swapped back in the dirinv routines)
+        out_dims_xyh = (*self._other_dims, "xlon", "ylat", "height")
+        out_shape_xyh = (*self._other_shape, _field.xlon.size, _nlat, _field.height.size)
+        # Combine all outputs into a dataset, reshape to restore the original
+        # other dimensions that were flattened earlier
+        return xr.Dataset(
+            data_vars={
+                "astarbaro": (out_dims_xy, out_fields["astarbaro"].reshape(out_shape_xy)),
+                "ubaro": (out_dims_xy, out_fields["ubaro"].reshape(out_shape_xy)),
+                "urefbaro": (out_dims_y, out_fields["urefbaro"].reshape(out_shape_y)),
+                "ua1baro": (out_dims_xy, out_fields["ua1baro"].reshape(out_shape_xy)),
+                "ua2baro": (out_dims_xy, out_fields["ua2baro"].reshape(out_shape_xy)),
+                "ep1baro": (out_dims_xy, out_fields["ep1baro"].reshape(out_shape_xy)),
+                "ep2baro": (out_dims_xy, out_fields["ep2baro"].reshape(out_shape_xy)),
+                "ep3baro": (out_dims_xy, out_fields["ep3baro"].reshape(out_shape_xy)),
+                "ep4": (out_dims_xy, out_fields["ep4"].reshape(out_shape_xy)),
+                "astar1": (out_dims_xyh, out_fields["astar1"].reshape(out_shape_xyh)),
+                "astar2": (out_dims_xyh, out_fields["astar2"].reshape(out_shape_xyh)),
+            },
+            coords={
+                **self._other_coords,
+                "height": _field.height,
+                "ylat": _field.ylat[-_nlat:],
+                "xlon": _field.xlon,
+            },
+            attrs=self.attrs
+        )
+
+
+def _is_equator(x):
+    return abs(x) < 1.0e-4
+
+def hemisphere_to_globe(ds, var_names=None):
```

Comment:
Thanks for creating this function! In the current numpy version of code, the wave activity fluxes for southern hemisphere is also computed this way actually. I shall think about how to generalize such operation (mirroring across equator) to be used at multiple places.